src/error.rs

```diff
@@ -16,6 +15,162 @@ use rustls::{CertificateError, Error, InvalidMessage};
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub struct rustls_io_result(pub libc::c_int);
 
+macro_rules! u32_enum_builder {
+    (
+    $(#[$comment:meta])*
+        EnumName: $enum_name: ident;
+        EnumDefault: $enum_default: ident;
+        EnumVal { $( $enum_var: ident => $enum_val: expr ),* }
+    ) => {
+        $(#[$comment])*
+        #[allow(dead_code)]
+        #[repr(u32)]
+        #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+        pub enum $enum_name {
+            $( $enum_var = $enum_val),*
+        }
+        impl From<u32> for $enum_name {
```

Comment:
another quick note to self before I drop: I'm surprised I didn't need to implement `TryFrom`. Merits a deeper look into why.

Reply:
Explained in https://github.com/rustls/rustls-ffi/pull/322#issuecomment-1613805427 and https://github.com/rustls/rustls-ffi/pull/322#issuecomment-1613808862